Expanding language about what can be animated

PatrickRyanMS opened the pull request:
Our animation docs were very mesh-centric. Expanded language to include non-mesh objects that can be animated. I added @deltakosh to this review to ensure that the nuance of the language updated here is not inaccurate or misleading.

Closes #984

PatrickRyanMS committed:
Expanding language about what can be animated

Our animation docs were very mesh-centric. Expanded language to include non-mesh objects that can be animated.

carolhmj reviewed (commented):
Great updates! Just have a few nits suggestions on wording 😄

Popov72 reviewed (approved):
LGTM!

RaananW commented:
Let me know if this can be merged. There are two open suggestions

PatrickRyanMS committed:
Update content/features/featuresDeepDive/animation/animation_design.md

Co-authored-by: Carolina <carolhmj@gmail.com>

PatrickRyanMS committed:
Update content/features/featuresDeepDive/animation/animation_design.md

Co-authored-by: Carolina <carolhmj@gmail.com>

PatrickRyanMS committed:
Update animation_method.md

added clarification to beginAnimation to address animations stored in target.animations

PatrickRyanMS commented:
> Let me know if this can be merged. There are two open suggestions

I merged the suggestions from @carolhmj and added additional context to cover the suggestion from @Popov72. If @Popov72 is good with the addition, we should be good to merge.
